src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -1907,6 +1907,72 @@ class MavlinkStreamGPS2Raw : public MavlinkStream
 	}
 };
 
+class MavlinkStreamGPSStatus : public MavlinkStream
+{
+public:
+	const char *get_name() const override
+	{
+		return MavlinkStreamGPSStatus::get_name_static();
+	}
+
+	static constexpr const char *get_name_static()
+	{
+		return "GPS_STATUS";
+	}
+
+	static constexpr uint16_t get_id_static()
+	{
+		return MAVLINK_MSG_ID_GPS_STATUS;
+	}
+
+	uint16_t get_id() override
+	{
+		return get_id_static();
+	}
+
+	static MavlinkStream *new_instance(Mavlink *mavlink)
+	{
+		return new MavlinkStreamGPSStatus(mavlink);
+	}
+
+	unsigned get_size() override
+	{
+		return _satellite_info_sub.advertised() ? MAVLINK_MSG_ID_GPS_STATUS_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES : 0;
+	}
+
+private:
+	uORB::Subscription _satellite_info_sub{ORB_ID(satellite_info)};
+
+	/* do not allow top copying this class */
+	MavlinkStreamGPSStatus(MavlinkStreamGPSStatus &) = delete;
+	MavlinkStreamGPSStatus &operator = (const MavlinkStreamGPSStatus &) = delete;
+
+protected:
+	explicit MavlinkStreamGPSStatus(Mavlink *mavlink) : MavlinkStream(mavlink)
+	{}
+
+	bool send(const hrt_abstime t) override
+	{
+		satellite_info_s sat {};
+
+		if (_satellite_info_sub.update(&sat)) {
+			mavlink_gps_status_t msg{};
+
+			memcpy(&msg.satellite_used[0], &sat.used, sizeof(msg.satellite_used));
```

Comment:
Please ensure the size matches: `static_assert(sizeof(msg.satellite_used) == sizeof(sat.used), "gps message size mismatch");`

Reply:
Thanks for pointing this out, I've refactored it, commits to follow momentarily.